c/serverStatusService.c

```diff
@@ -0,0 +1,284 @@
+    printf("Failed to fetch RMF data, RC = %d\n", rc);
+    return;
+  }
+  time_t ltime;
+  char tstamp[50];
+  time(&ltime);
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  if(ctime_r(&ltime, tstamp) != NULL){
+    if(tstamp[strlen(tstamp) - 1] != '\0'){
+      tstamp[strlen(tstamp) - 1] = '\0';
+    }
+    jsonAddString(out, "timestamp", tstamp);
+  }
+  if(getenv("ZSS_LOG_FILE") != NULL){
+    jsonAddString(out, "logDirectory", getenv("ZSS_LOG_FILE"));
+  } else {
+    jsonAddString(out, "logDirectory", "ZSS_LOG_FILE not defined in environment variables");
+  }
+  jsonAddString(out, "agentName", "zss");
+  jsonAddString(out, "agentVersion", context->productVersion);
+  jsonAddString(out, "arch", unameRet.sysname);
+  jsonAddString(out, "osRelease", unameRet.release);
+  jsonAddString(out, "osVersion", unameRet.version);
+  jsonAddString(out, "hardwareIdentifier", unameRet.machine);
+  jsonAddString(out, "hostname", hostnameBuffer);
+  jsonAddString(out, "nodename", unameRet.nodename);
+  jsonStartObject(out, "userEnvironment");
+  char *envVar = *environ;
+  for (int i = 1; envVar; i++) {
+    char *equalSign = strchr(envVar, '=');
+    if(equalSign == NULL){
+      break;
+    }
+    int nameLen = strlen(envVar) - strlen(equalSign);
+    int nameBufferLen = nameLen + 1;
+    char *name = safeMalloc(nameBufferLen, "env_var name");
+    if(name == NULL){
+      break;
+    }
+    memcpy(name, envVar, nameLen);
+    name[nameLen] = '\0';
+    jsonAddString(out, name, equalSign+1);
+    safeFree(name, nameBufferLen);
+    envVar = *(environ+i);
+  }
+  jsonEndObject(out);
+  jsonAddInt(out, "demandPagingRate", demandPaging);
+  jsonAddInt(out, "stdCP_CPU_Util", cpuUtil);
+  jsonAddInt(out, "stdCP_MVS_SRM_CPU_Util", mvsSrm);
+  jsonAddInt(out, "ZAAP_CPU_Util", zaapUtil);
+  jsonAddInt(out, "ZIIP_CPU_Util", ziipUtil);
+  jsonAddInt(out, "PID", getpid());
+  jsonAddInt(out, "PPID", getppid());
+  jsonEnd(out);
+  finishResponse(response);
+}
+
+static int serveStatus(HttpService *service, HttpResponse *response) {
+  HttpRequest *request = response->request;
+  ServerAgentContext *context = service->userPointer;
+  //This service is conditional on RBAC being enabled because it is a 
+  //sensitive URL that only RBAC authorized users should be able to access
+  JsonObject *dataserviceAuth = jsonObjectGetObject(context->serverConfig, "dataserviceAuthentication");
+  int rbacParm = jsonObjectGetBoolean(dataserviceAuth, "rbac");
+  if(!rbacParm){
+     respondWithError(response, HTTP_STATUS_BAD_REQUEST, "Set dataserviceAuthentication.rbac to true in server configuration");
+  } else {
+    if(!strcmp(request->method, methodGET)){
+      char *l1 = stringListPrint(request->parsedFile, 2, 1, "/", 0);
+      if(!strcmp(l1, "")){
+        respondWithServerRoutes(response);
+      }else if (!strcmp(l1, "config")){
```

Comment:
I think we should be more consistent with formatting. Please use spaces after in argument lists and before/after brackets.

For example,
```c
if(!strcmp(l1, "")){
}else if (!strcmp(l1, "config")){
 }else if (!strcmp(l1, "log")) {
} else {
```
should be
```c
if (!strcmp(l1, "")) {
} else if (!strcmp(l1, "config")) {
 } else if (!strcmp(l1, "log")) {
} else {
```